src/utils/breakdown_validation.py

```diff
@@ -115,54 +148,224 @@ def greater_than_validation(
-        pd.DataFrame
+        dict: A dictionary with boolean values for each check.
+        pd.DataFrame: The dataframe with the breakdown errors
     """
+    qa_df = df.copy()
+    wanted_refs = []  # a list of references that have errors
+    cols = []  # a list of columns that have errors
 
-    if check == 'constructed':
-        BreakdownValidationLogger.info("Running breakdown validation for constructed data")
-        validation_df = df[df['is_constructed'] == True].copy()
-        not_for_validating_df = df[df['is_constructed'] == False].copy()
-    else:
-        validation_df = df.copy()
+    check_results_dict = {}
+    for key, columns in to_check.items():
+        total_column = columns[-1]
+        breakdown_columns = columns[:-1]
+        check_cond = abs(df[breakdown_columns].sum(axis=1) - df[total_column]) > 0.005
+        # if there are any errors for particular check..
+        if any(check_cond):
+            # ...create a mini dataframe for the relevant rows and columns
+            wanted_cols = ["reference", "instance", "imp_class", "imp_marker"] + list(
+                columns
+            )
+            check_df = df.copy().loc[check_cond, wanted_cols]
+            check_df[f"{key}_diff"] = (
+                df.loc[check_cond, breakdown_columns].sum(axis=1)
+                - df.loc[check_cond, total_column]
+            )
 
-    validation_df = replace_nulls_with_zero(validation_df)
-    rows_to_validate = remove_all_nulls_rows(validation_df)
-    msg, count = equal_validation(rows_to_validate)
-    msg, count = greater_than_validation(rows_to_validate, msg, count)
+            # add the diff column to the qa dataframe
+            qa_df.loc[check_cond, f"{key}_diff"] = check_df[f"{key}_diff"]
+            wanted_refs += [
+                r for r in (check_df["reference"].tolist()) if r not in wanted_refs
+            ]
+            cols += [c for c in check_df.columns if c not in cols] + [f"{key}_diff"]
+
+            # add the mini-dataframe to the dict for unit testing and logger messages
+            check_results_dict[key] = check_df
+        else:
+            check_results_dict[key] = pd.DataFrame()
+
+    # Filter and select so the qa dataframe has only relevant columns and rows
+    qa_df = qa_df.loc[df["reference"].isin(wanted_refs)][cols]
+
+    return check_results_dict, qa_df
+
+
+def log_errors_to_screen(check_results_dict: dict, check_type: str) -> None:
+    """Function to log the errors to the screen.
 
-    if check != 'all':
-        df = pd.concat([validation_df, not_for_validating_df], ignore_index=True)
+    Args:
+        check_results_dict (dict): The dictionary of errors to log.
+
+    Returns:
+        None
+    """
+    for key, value in check_results_dict.items():
+        if not value.empty:
+            BreakdownValidationLogger.error(
+                f"Breakdown validation failed for {key} columns"
+            )
+            BreakdownValidationLogger.error(value)
     else:
```

Comment:
Indentation of the else - it's currently logging the errors and the 'all vals are valid' msg